CI 1.22.x rns (#6186)
harness/developer-hub: release-notes/continuous-integration.md

* Potentially empty output variables populating environment variables in child stages.
* [Bitrise Workflow Steps](/docs/continuous-integration/use-ci/use-drone-plugins/ci-bitrise-plugin) requiring empty environment variables.

For more information about CI early access features, go to [What's supported for Harness CI](/docs/continuous-integration/ci-supported-platforms).

#### Fixed issues

* Builds triggered by Bitbucket Server push events now have correct date information in the build history. This issue occurred due to missing date information in the `commits` object returned by the Bitbucket Server API. This change requires Harness Delegate version 24.04.XXXXX or later. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate). (CI-11556, ZD-58798)
* Fixed an issue where references to deleted/nonexistent secrets sometimes caused pipelines to timeout at the initialize step without any logs. (CI-11891, ZD-60575)

> * Added a fix to trim long environment variables, such as the commit message and PR title, during pod creation in stage initialization. This is currently an opt-in fix for effected customers. If you are using a Kubernetes cluster build infrastructure and experiencing initialization timeout with an error message like `rpc error: code = ResourceExhausted desc = trying to send message larger than max`, contact [Harness Support](mailto:support@harness.io) to enable this fix in your account. (CI-11709, ZD-59521)

Reviewer: ```suggestion
* Added a fix to trim long environment variables, such as the commit message and PR title, during pod creation in stage initialization. This is currently an opt-in fix for affected customers. If you're using a Kubernetes cluster build infrastructure and experiencing initialization timeout with an error message like `rpc error: code = ResourceExhausted desc = trying to send message larger than max`, contact [Harness Support](mailto:support@harness.io) to enable this fix in your account. (CI-11709, ZD-59521)
```

## April 2024

### Version 1.22.x

<!-- 09 Apr 2024 -->

#### Early access features

> The feature flag `CI_USE_LESS_STRICT_EVALUATION_FOR_MAP_VARS` can resolve some issues where certain variable expressions aren't correctly resolved. Possible use cases include some variables related to looping strategies that incorrectly resolve to null if included in environment variable expressions or empty output variables populating environment variables in child stages. For more information about CI early access features, go to [What's supported for Harness CI](/docs/continuous-integration/ci-supported-platforms). (CI-11882, CI-11305 ZD-57626)

Reviewer: ```suggestion
The feature flag `CI_USE_LESS_STRICT_EVALUATION_FOR_MAP_VARS` can resolve some issues where certain variable expressions aren't correctly resolved. (CI-11882, CI-11305 ZD-57626)

Here are some issues where this feature flag might be useful:
- Some variables related to looping strategies incorrectly resolve to null if included in environment variable expressions.
- Empty output variables populate environment variables in child stages. 

For more information about CI early access features, go to [What's supported for Harness CI](/docs/continuous-integration/ci-supported-platforms). 
```

#### Fixed issues

> Builds triggered by Bitbucket Server push events have correct date information in the build history. This issue occurred due to missing date information in the `commits` object returned by the Bitbucket Server API. This change requires Harness Delegate version 24.04.XXXXX or later. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate). (CI-11556, ZD-58798)

Reviewer: ```suggestion
Builds triggered by Bitbucket Server push events now have correct date information in the build history. This issue occurred due to missing date information in the `commits` object returned by the Bitbucket Server API. This change requires Harness Delegate version 24.04.XXXXX or later. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate). (CI-11556, ZD-58798)
```
